server/routes/companies.js

```diff
@@ -116,25 +179,459 @@ router.get('/filter', function(req, res, next) {
+    } else {
+        companyEvents.push({
+           event_id: event_ids
+        })
+    }
+  }
+
+
+  var queryCompany = knex('company').insert(values)
+  knex.transaction(function(trx) {
+    return queryCompany.transacting(trx)
+      .returning('id')
+      .then(async function(ids){
+        let company_id = ids[0];
+
+        if (ranks && user_ids) {
+          rankValues.forEach(function(element){
+            element.company_id = company_id;
+          })
+          var queryRank = knex('user_company_rank').insert(rankValues)
+          await queryRank.transacting(trx);
+        }
+
+        if (position_ids) {
+          companyPositions.forEach(function(element){
+            element.company_id = company_id;
+          })
+          var queryPos = knex('company_position').insert(companyPositions)
+          await queryPos.transacting(trx);
+        }
+
+        if (major_ids) {
+          companyMajors.forEach(function(element){
+            element.company_id = company_id;
+          })
+          var queryMajor = knex('company_major').insert(companyMajors)
+          await queryMajor.transacting(trx);
+        }
+
+        if (contact_ids) {
+          companyContacts.forEach(function(element){
+            element.company_id = company_id;
+          })
+          var queryContact = knex('company_contact').insert(companyContacts)
+          await queryContact.transacting(trx);
+        }   
+
+        if (event_ids) {
+          companyEvents.forEach(function(element){
+            element.company_id = company_id;
+          })
+          var queryEvent = knex('event_company').insert(companyEvents)
+          await queryEvent.transacting(trx);
+        }
+
+        return trx.commit;
+      })
+    })
+    .then(function(){
+      res.send(util.message('Successfully inserted company'));
+    })
+    .catch(err => {return next(err) });
+  });
+
+// Update a single company
+router.put('/:company_id', function(req, res, next) {
+  values = {
+    name: req.body.name,
+    website: req.body.website,
+    logo: req.body.logo,
+    citizenship_requirement: req.body.citizenship_requirement,
+    description: req.body.description
+  }
+  let ranks = req.body.rank;
+  let user_ids = req.body.user_id;
+  let position_ids = req.body.position_id;
+  let major_ids = req.body.major_id;
+  let contact_ids = req.body.contact_id;
+  let event_ids = req.body.event_id;
+
+  let remove_user_ids = req.body.remove_user_id;
+  let remove_position_ids = req.body.remove_position_id;
+  let remove_major_ids = req.body.remove_major_id;
+  let remove_contact_ids = req.body.remove_contact_id;
+  let remove_event_ids = req.body.remove_event_id;
+  let company_id = req.params.company_id;
+
+  //insert
+  let rankValues = [];
+  if (ranks && user_ids) {
+    if(Array.isArray(ranks) && Array.isArray(user_ids)){
+      let length = ranks.length;
+      if (ranks.length > user_ids.length)
+        length = user_ids.length; //not sure about error handling for length
```

Comment:
I. Am. So. Sorry. I wasn't thinking clearly but I now see why you have this user/ranks stuff. The user_company_rank table will actuallly be part of the USERS API, not the companies API lol. Since we'd logically have users with company preferences, not from the perspective of companies. You can just remove any code you have for the company rank stuff... sorry!!! (or send to nikhita for reference if she needs it!!!)

![image](https://user-images.githubusercontent.com/17267100/57702100-2e501580-7612-11e9-8f5f-d29a450a571a.png)

Reply:
got it! I sent Nikhita the relevant code snippets if she needs them